### PR TITLE
Port over ES OpenCloseIndexIT tests.

### PR DESCRIPTION
**Pull request by kovrus**
## Summary of the changes / Why this improves CrateDB

ES state: `704317da71c`

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java [hunk 1]
```diff
@@ -40,6 +45,160 @@ public void prepareClosedTable() {
         execute("alter table t close");
     }
 
+    @UseJdbc(1)
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java:48]: Is the `UseJdbc(1)` required here?

File: server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java [hunk 2]
```diff
@@ -40,6 +45,160 @@ public void prepareClosedTable() {
...
+        assertThat(response.rows()[0][0], is(numberOfDocs));
+        assertThat(response.rows()[0][1], is(0L));
+    }
+
+    @Test
+    public void test_open_close_is_not_blocked_with_read_and_write_blocks_enabled() {
+        execute("create table test (x int) with (number_of_replicas = 0) ");
+        for (int i = 0; i < between(10, 20); i++) {
+            execute("insert into test values (?)", new Object[]{i});
+        }
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java:129]: Might be worth changing this to a single bulk insert with `between(10, 20)` to speed up the test case a bit.

File: server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java [hunk 3]
```diff
@@ -40,6 +45,160 @@ public void prepareClosedTable() {
...
+            }
+        }
+    }
+
+    @Test
+    public void test_close_is_blocked_with_read_only_and_metadata_blocks_enabled() {
+        execute("create table test (x int) with (number_of_replicas = 0) ");
+        for (int i = 0; i < between(10, 20); i++) {
+            execute("insert into test values (?)", new Object[]{i});
+        }
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java:155]: same here

File: server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java [hunk 4]
```diff
@@ -40,6 +45,160 @@ public void prepareClosedTable() {
...
+            }
+        }
+    }
+
+    @Test
+    public void test_open_is_blocked_with_read_and_write_blocks_enabled() {
+        execute("create table test (x int) with (number_of_replicas = 0) ");
+        for (int i = 0; i < between(10, 20); i++) {
+            execute("insert into test values (?)", new Object[]{i});
+        }
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java:175]: and here

**Review by mfussenegger — APPROVED** [review]
Left a question and a suggestion, otherwise lgtm :+1:

File: server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java (hunk 1 above)
Comment (kovrus) [reply server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java]: it is not needed, I used the wrong crate error code + HTTP code, so the assertion worked only for the pg but not for HTTP case.

**Pull request merged**